pkg/apis/v1beta1/nodepool_validation_cel_test.go

```diff
@@ -97,4 +99,217 @@ var _ = Describe("CEL/Validation", func() {
+			It("should succeed on evictionSoft with valid keys", func() {
+				nodePool.Spec.Template.Spec.Kubelet = &KubeletConfiguration{
+					EvictionSoft: map[string]EvictionValue{
+						"memory.available":   "5%",
+						"nodefs.available":   "10%",
+						"nodefs.inodesFree":  "15%",
+						"imagefs.available":  "5%",
+						"imagefs.inodesFree": "5%",
+						"pid.available":      "5%",
+					},
+					EvictionSoftGracePeriod: map[string]metav1.Duration{
+						"memory.available":   {Duration: time.Minute},
+						"nodefs.available":   {Duration: time.Second * 90},
+						"nodefs.inodesFree":  {Duration: time.Minute * 5},
+						"imagefs.available":  {Duration: time.Hour},
+						"imagefs.inodesFree": {Duration: time.Hour * 24},
+						"pid.available":      {Duration: time.Minute},
+					},
+				}
+				Expect(env.Client.Create(ctx, nodePool)).To(Succeed())
+			})
+			It("should fail on evictionSoft with invalid keys", func() {
+				nodePool.Spec.Template.Spec.Kubelet = &KubeletConfiguration{
+					EvictionSoft: map[string]EvictionValue{
+						"memory": "5%",
+					},
+					EvictionSoftGracePeriod: map[string]metav1.Duration{
+						"memory": {Duration: time.Minute},
+					},
+				}
+				Expect(env.Client.Create(ctx, nodePool)).ToNot(Succeed())
+			})
+			It("should fail on invalid formatted percentage value in evictionSoft", func() {
+				nodePool.Spec.Template.Spec.Kubelet = &KubeletConfiguration{
+					EvictionSoft: map[string]EvictionValue{
+						"memory.available": "5%3",
+					},
+					EvictionSoftGracePeriod: map[string]metav1.Duration{
+						"memory.available": {Duration: time.Minute},
+					},
+				}
+				Expect(env.Client.Create(ctx, nodePool)).ToNot(Succeed())
+			})
+			It("should fail on invalid percentage value (too large) in evictionSoft", func() {
+				nodePool.Spec.Template.Spec.Kubelet = &KubeletConfiguration{
+					EvictionSoft: map[string]EvictionValue{
+						"memory.available": "110%",
+					},
+					EvictionSoftGracePeriod: map[string]metav1.Duration{
+						"memory.available": {Duration: time.Minute},
+					},
+				}
+				Expect(env.Client.Create(ctx, nodePool)).ToNot(Succeed())
+			})
+			It("should fail on invalid quantity value in evictionSoft", func() {
+				nodePool.Spec.Template.Spec.Kubelet = &KubeletConfiguration{
+					EvictionSoft: map[string]EvictionValue{
+						"memory.available": "110GB",
+					},
+					EvictionSoftGracePeriod: map[string]metav1.Duration{
+						"memory.available": {Duration: time.Minute},
+					},
+				}
+				Expect(env.Client.Create(ctx, nodePool)).ToNot(Succeed())
+			})
+			It("should fail when eviction soft doesn't have matching grace period", func() {
+				nodePool.Spec.Template.Spec.Kubelet = &KubeletConfiguration{
+					EvictionSoft: map[string]EvictionValue{
+						"memory.available": "200Mi",
+					},
+				}
+				Expect(env.Client.Create(ctx, nodePool)).ToNot(Succeed())
+			})
+		})
+		Context("GCThresholdPercent", func() {
+			Context("ImageGCHighThresholdPercent", func() {
```

Comment:
Can we collapse these two contexts together? I think we could just do 3 statements that validate the following:
1. "should succeed on valid GCThresholdPercents"
2. "should fail when imageGCHighThresholdPercent is less than imageGCLowThresholdPercent" 
3. "should fail when imageGCLowThresholdPercent is greather than imageGCHighThresheldPercent"